source/fundamentals/connection/connection-options.txt

```diff
@@ -6,203 +6,290 @@ Connection Options
+       This count includes connections in use.
+
+   * - **proxyHost**
      - string
      - ``null``
-     - Specifies the logger level used by the driver. Valid choices are:
-       ``error``, ``warn``, ``info``, and ``debug``.
+     - Specifies the IPv4/IPv6 address or domain name of a SOCKS5 proxy
+       server used for connecting to MongoDB services.
 
-   * - **maxConnecting**
-     - integer
-     - ``2``
-     - Specifies the maximum number of connections a connection pool can establish 
-       concurrently. 
+   * - **proxyPort**
+     - non-negative integer
+     - ``null``
+     - Specifies the port of the SOCKS5 proxy server specified in ``proxyHost``.
 
-   * - **maxIdleTimeMS**
-     - integer
-     - ``∞``
-     - Specifies the maximum amount of time a connection can remain idle in the
-       connection pool before being removed and closed.
+   * - **proxyUsername**
+     - string
+     - ``null``
+     - Specifies the username for username/password authentication to the SOCKS5
+       proxy server specified in ``proxyHost``.
 
-   * - **maxPoolSize**
-     - integer
-     - ``100``
-     - Specifies the maximum number of connections that a connection pool may
-       have at a given time.
+   * - **proxyPassword**
+     - string
+     - ``null``
+     - Specifies the password for username/password authentication to the SOCKS5
+       proxy server specified in ``proxyHost``.
 
-   * - **minPoolSize**
-     - integer
-     - ``0``
-     - Specifies the minimum number of connections that must exist at any moment
-       in a single connection pool.
+   * - **readConcernLevel**
+     - string
+     - ``null``
+     - Specifies the default read concern for the client. See :ref:`read concern
+       <read-concern>` for more information.
 
-   * - **noDelay**
-     - boolean
-     - ``true``
-     - Specifies whether to use the TCP socket no-delay option. For more
-       information, see the documentation for `Node.js socket.setNoDelay
-       <https://nodejs.org/dist/latest-v10.x/docs/api/net.html#net_socket_setnodelay_nodelay>`_.
+   * - **readPreference**
+     - string
+     - ``"primary"``
+     - Specifies the default read preference for the client (excluding tags). See :ref:`read preference <read-preference>` for more information.
 
-   * - **pkFactory**
-     - object
+   * - **readPreferenceTags**
+     - comma-separated key:value pairs (e.g. "dc:ny,rack:1" and "dc:ny)
+       can be specified multiple times, each instance of this key is a
+       separate tag set
      - ``null``
-     - Specifies a primary key factory object that generates custom ``_id``
-       keys.
+     - Specifies the default read preference tags for the client. This option is
+       valid only if the read preference mode is not primary
 
-   * - **promiseLibrary**
-     - object
+       The order of the tag sets in the read preference is the same as the order
+       they are specified in the URI.
```

Comment:
Issue: This was hard for me to understand (had to read it 3-4 times and still unsure of meaning). could this be clarified? Alright if this is ticketed out separately.